public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -133,7 +138,8 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
           data: [lokiResultsToTableModel(response.data.data.result, responseListLength, target.refId, meta, true)],
           key: `${target.refId}_instant`,
         };
-      })
+      }),
+      catchError((err: any) => this.throwUnless(err, err.status === 404, target))
```

Comment:
Why is this needed? Is this for legacy loki instances that don't have the query endpoint?

Reply:
It in not necessary, but in [throwUnless](https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/loki/datasource.ts#L505-L520) we have a function that enriches the error message to give users a better guidance on errors (specifically escaping error).

Reply:
So instead of having this error message: 
![image](https://user-images.githubusercontent.com/30407135/91296105-5e3a1280-e79c-11ea-947b-1f0f306f5e3d.png)

We get this - more guiding - error message:
![image](https://user-images.githubusercontent.com/30407135/91296049-495d7f00-e79c-11ea-9c62-6a637aea4825.png)